Lib/pyclbr.py

```diff
@@ -52,40 +53,41 @@
 
 class _Object:
     "Information about Python class or function."
-    def __init__(self, module, name, file, lineno, parent):
+    def __init__(self, module, name, file, lineno, end_lineno, parent):
         self.module = module
         self.name = name
         self.file = file
         self.lineno = lineno
+        self.end_lineno = end_lineno
         self.parent = parent
         self.children = {}
         if parent is not None:
             parent.children[name] = self
 
 class Function(_Object):
     "Information about a Python function, including methods."
-    def __init__(self, module, name, file, lineno, parent=None, is_async=False):
-        super().__init__(module, name, file, lineno, parent)
+    def __init__(self, module, name, file, lineno, end_lineno, parent=None, is_async=False):
+        super().__init__(module, name, file, lineno, end_lineno, parent)
         self.is_async = is_async
         if isinstance(parent, Class):
             parent.methods[name] = lineno
 
 class Class(_Object):
     "Information about a Python class."
-    def __init__(self, module, name, super_, file, lineno, parent=None):
-        super().__init__(module, name, file, lineno, parent)
+    def __init__(self, module, name, super_, file, lineno, end_lineno, parent=None):
+        super().__init__(module, name, file, lineno, end_lineno, parent)
         self.super = super_ or []
         self.methods = {}
 
 # These 2 functions are used in these tests
 # Lib/test/test_pyclbr, Lib/idlelib/idle_test/test_browser.py
-def _nest_function(ob, func_name, lineno, is_async=False):
+def _nest_function(ob, func_name, lineno, end_lineno, is_async=False):
     "Return a Function after nesting within ob."
-    return Function(ob.module, func_name, ob.file, lineno, ob, is_async)
+    return Function(ob.module, func_name, ob.file, lineno, end_lineno, ob, is_async)
```

Comment:
```suggestion
    return Function(ob.module, func_name, ob.file, lineno, ob, is_async, end_lineno=end_lineno)
```
Same or equivalent change for other 3 Function/Class calls.